src/Hl7.Fhir.Base/Serialization/OperationOutcomeExtensions.cs

```diff
@@ -0,0 +1,140 @@
+                case FhirJsonException.NO_RESOURCETYPE_PROPERTY_CODE: shortDisplay = "Missing ResourceType"; break;
+                case FhirJsonException.EXPECTED_PRIMITIVE_NOT_OBJECT_CODE: shortDisplay = "Expected Object"; break;
+                case FhirJsonException.EXPECTED_PRIMITIVE_NOT_ARRAY_CODE: shortDisplay = "Expected Primitive"; break;
+                case FhirJsonException.INCORRECT_BASE64_DATA_CODE: shortDisplay = "Invalid Base64 data"; break;
+                // case FhirJsonException.STRING_ISNOTAN_INSTANT_CODE: shortDisplay = "JSON107"; break;
+                case FhirJsonException.NUMBER_CANNOT_BE_PARSED_CODE: shortDisplay = "Invalid number"; break;
+                // case FhirJsonException.EXPECTED_PRIMITIVE_NOT_NULL_CODE: shortDisplay = "JSON109"; break;
+                case FhirJsonException.UNEXPECTED_JSON_TOKEN_CODE: shortDisplay = "Invalid datatype"; break;
+                // case FhirJsonException.EXPECTED_START_OF_ARRAY_CODE: shortDisplay = "JSON111"; break;
+                // case FhirJsonException.USE_OF_UNDERSCORE_ILLEGAL_CODE: shortDisplay = "JSON113"; break;
+                // case FhirJsonException.CHOICE_ELEMENT_HAS_NO_TYPE_CODE: shortDisplay = "JSON114"; break;
+                // case FhirJsonException.CHOICE_ELEMENT_HAS_UNKOWN_TYPE_CODE: shortDisplay = "JSON115"; break;
+                case FhirJsonException.UNKNOWN_RESOURCE_TYPE_CODE: shortDisplay = "Invalid ResourceType"; break;
+                // case FhirJsonException.RESOURCE_TYPE_NOT_A_RESOURCE_CODE: shortDisplay = "JSON117"; break;
+                case FhirJsonException.UNKNOWN_PROPERTY_FOUND_CODE: shortDisplay = "Unknown element"; break;
+                // case FhirJsonException.RESOURCETYPE_UNEXPECTED_CODE: shortDisplay = "JSON119"; break;
+                // case FhirJsonException.OBJECTS_CANNOT_BE_EMPTY_CODE: shortDisplay = "JSON120"; break;
+                // case FhirJsonException.ARRAYS_CANNOT_BE_EMPTY_CODE: shortDisplay = "JSON121"; break;
+                // case FhirJsonException.LONG_CANNOT_BE_PARSED_CODE: shortDisplay = "JSON122"; break;
+                // case FhirJsonException.LONG_INCORRECT_FORMAT_CODE: shortDisplay = "JSON123"; break;
+
+                // case FhirJsonException.PRIMITIVE_ARRAYS_ONLY_NULL_CODE: shortDisplay = "JSON125"; break;
+                // case FhirJsonException.INCOMPATIBLE_SIMPLE_VALUE_CODE: shortDisplay = "JSON126"; break;
+
+
+                case FhirXmlException.EMPTY_ELEMENT_NAMESPACE_CODE: shortDisplay = "Empty Element Namespace"; break;
+                case FhirXmlException.UNKNOWN_RESOURCE_TYPE_CODE: shortDisplay = "Invalid ResourceType"; break;
+                // case FhirXmlException.RESOURCE_TYPE_NOT_A_RESOURCE_CODE: shortDisplay = "XML103"; break;
+                case FhirXmlException.UNKNOWN_ELEMENT_CODE: shortDisplay = "Unknown element"; break;
+                // case FhirXmlException.CHOICE_ELEMENT_HAS_NO_TYPE_CODE: shortDisplay = "XML105"; break;
```

Comment:
Are these still todo?

Reply:
I'm happy to leave these at that, the others I'm not so fussed on as unlikely to have been a result of user input, but from bad software.